src/main/java/com/icthh/xm/ms/configuration/repository/impl/IdpConfigRepository.java

```diff
@@ -0,0 +1,138 @@
+package com.icthh.xm.ms.configuration.repository.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import com.icthh.xm.commons.domain.idp.IdpConfigUtils;
+import com.icthh.xm.commons.domain.idp.model.IdpPublicConfig;
+import com.icthh.xm.commons.domain.idp.model.IdpPublicConfig.IdpConfigContainer.IdpPublicClientConfig;
+import com.icthh.xm.ms.configuration.service.JwksService;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.util.CollectionUtils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.icthh.xm.commons.domain.idp.IdpConstants.IDP_PUBLIC_SETTINGS_CONFIG_PATH_PATTERN;
+
+/**
+ * This class reads and process IDP clients public configuration for each tenant.
+ * Tenant IDP clients created for each successfully loaded config. If config not loaded or invalid it skipped.
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class IdpConfigRepository implements RefreshableConfiguration {
+
+    public static final String KEY_TENANT = "tenant";
+
+    private final JwksService jwksService;
+
+    private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    private final AntPathMatcher matcher = new AntPathMatcher();
+
+    /**
+     * In memory storage for storing information tenant IDP clients public.
+     */
+    private final Map<String, Map<String, IdpPublicClientConfig>> idpClientConfigs = new ConcurrentHashMap<>();
+
+    /**
+     * In memory storage.
+     * Stores information about tenant IDP clients public configuration that currently in process.
+     * <p/>
+     * We need to store this information in memory to avoid corruption previously registered in-memory tenant clients config
+     */
+    private final Map<String, Map<String, IdpPublicClientConfig>> tmpIdpClientPublicConfigs = new ConcurrentHashMap<>();
+
+    @Override
+    public boolean isListeningConfiguration(String updatedKey) {
+        return matcher.match(IDP_PUBLIC_SETTINGS_CONFIG_PATH_PATTERN, updatedKey);
+    }
+
+    @Override
+    public void onInit(String configKey, String configValue) {
+        updateIdpConfigs(configKey, configValue);
+    }
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+        updateIdpConfigs(updatedKey, config);
+    }
+
+    private void updateIdpConfigs(String configKey, String config) {
```

Comment:
pls handle case when config removed, in this case config will be null, and it's case must be handled and all information need to be deleted from memory

Reply:
Deletion case handled